### DISCUSSION TITLE
Face Power issue Data Center side

### DISCUSSION THREAD

**Discussion by gprao08** [Sunbird-cQube/community#870] [General]
@RNarmatha 
Hi Narmatha Good Morning
We have a power issue at our SDC side so that our VM shut down and stopped our cQube application . Later we restarted the VM but identify nginx container service exited. Screen shot attached (by _**docker ps -a command**_) for your reference. Could you plz provide steps to restart all container services.
![docker services_nginx](https://github.com/Sunbird-cQube/community/assets/143190553/6c572001-e821-4feb-b62c-09950428ecc3)

**Comment by RNarmatha** [comment]
Hi @rhwarrier- Could you please help with the next step to be taken here?

**Comment by PavitraIR** [comment]
@gprao08 
Nginx has been removed from the container. Please rebuild Nginx and verify all containers. please find the below command 
   
1.   cd cqube-devops/microservices/nginx-ms
2.  sudo docker build -t nginx_ms:1 .
3.    cd cqube-devops/ansible
4.    sudo docker-compose -f docker-compose.yml up -d

   
If the container still shows as exited after rebuilding Nginx, follow the steps outlined below.

1.    sudo docker stop nginx_app
2.    sudo docker rm nginx_app
3.    sudo docker rmi nginx_ms:1
5.    cd cqube-devops/microservices/nginx-ms
6.    sudo docker build -t nginx_ms:1 .
7.    cd cqube-devops/ansible
8.    sudo docker-compose -f docker-compose.yml up -d

**Reply by gprao08** [reply]
@PavitraIR 
followed the steps mentioned and it show nginx created. But cant reach the web site vsk.ap.gov.in. Could you plz support further to resolve the issue. Screen shot attached by _**docker ps -a**_
![image](https://github.com/Sunbird-cQube/community/assets/143190553/91eb793c-7c55-477c-a28d-cdfe066399b0)

**Reply by PavitraIR** [reply]
@gprao08 The container failed to start. Please ensure that the container is built correctly.find the below steps

1. sudo docker stop nginx_app
2. sudo docker rm nginx_app
3. sudo docker rmi nginx_ms:1
4. cd cqube-devops/microservices/nginx-ms
5. sudo docker build -t nginx_ms:1 .
6. cd cqube-devops/ansible
7. sudo docker-compose -f docker-compose.yml up -d

**Reply by gprao08** [reply]
@PavitraIR 
After giving sudo docker-compose -f docker-compose.yml up -d creating nginx app shows error. Screen shot attached for your reference.
Could you plz help us to resolve. 
![image](https://github.com/Sunbird-cQube/community/assets/143190553/f71fbb6a-7f4c-4874-a9fe-05b20a3b2dda)

**Reply by PavitraIR** [reply]
@gprao08 First check the certificate key and private.key and also check in default.old file whether ports are there or not, please find the below commands.

cd /opt
cd cqube
ls
cd nginix
ls
cat default.old

If any changes is needed means please modify it and save the file then restart the nginx container 

**sudo docker restart nginx_app**
  
**Note:** If everything is correct then also it is not working means it is from your server routing issue .

**Reply by gprao08** [reply]
found no certificate key and certificate (.crt) file in the below path
cd /opt/cqube/nginx/cert/vsk.ap.gov.in
Screen shot attached.
![image](https://github.com/Sunbird-cQube/community/assets/143190553/9ca184cd-17bd-4ee2-a4bc-2197472d6f6d)

Moreover there is only default.conf file and the consists the below data.

server {

        gzip on;
        gzip_types      application/javascript text/css;
        gzip_min_length 1000;
        gzip_static on;
        root /usr/share/nginx/html;
        server_tokens off;
        index index.html index.htm index.nginx-debian.html;

        add_header X-Frame-Options "SAMEORIGIN";
        add_header Strict-Transport-Security 'max-age=31536000; includeSubDomains; preload';
        add_header X-Content-Type-Options nosniff;
        add_header X-XSS-Protection "1; mode=block";
        add_header Content-Security-Policy "https://cdn.jsdelivr.net/npm/bootstrap@4.3.1/dist/css/bootstrap.min.css";
        add_header Referrer-Policy "strict-origin";
        add_header Permissions-Policy "geolocation=(),midi=(),sync-xhr=(),microphone=(),camera=(),magnetometer=(),gyroscope=(),fullscreen=(self),payment=()";

        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Host $host;
        proxy_set_header Host $host;
        proxy_set_header X-Forwarded-Proto https;
        #proxy_set_header X-Forwarded-Port $server_port;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_set_header Accept-Encoding "";        

        server_name vsk.ap.gov.in;

        location /dashboard {
                proxy_pass http://172.17.0.1:4200;
                #client_max_body_size 2G;
                #try_files $uri $uri/ =404;
        }
        location /admin {
                proxy_pass http://172.17.0.1:4201;
                #client_max_body_size 2G;
                #try_files $uri $uri/ =404;
        }
        location /api {
                proxy_pass http://172.17.0.1:8000;
                client_max_body_size 10G;
                #try_files $uri $uri/ =404;
        }
        location /grafana {
                proxy_pass http://172.17.0.1:3005;
                proxy_set_header Host $http_host;
                proxy_http_version 1.1;
                #try_files $uri $uri/ =404;
        }
        location /auth {
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header Host $host;
                proxy_pass http://172.17.0.1:8080$request_uri;
                proxy_set_header X-Forwarded-Proto https;
                proxy_set_header Accept-Encoding "";
        }
        location / {
                proxy_pass http://172.17.0.1:9000;
                proxy_set_header Host $http_host;
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header X-Forwarded-Proto $scheme;
                proxy_connect_timeout 300;
                # Default is HTTP/1, keepalive is only enabled in HTTP/1.1
                proxy_http_version 1.1;
                proxy_set_header Connection "";
                chunked_transfer_encoding off;
        }
        location /minio/ui/ {
                rewrite ^/minio/ui/(.*) /$1 break;
                proxy_set_header Host $http_host;
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header X-Forwarded-Proto $scheme;
                proxy_set_header X-NginX-Proxy true;
                # This is necessary to pass the correct IP to be hashed
                real_ip_header X-Real-IP;
                proxy_connect_timeout 300;
                proxy_http_version 1.1;
                proxy_set_header Upgrade $http_upgrade;
                proxy_set_header Connection "upgrade";
                chunked_transfer_encoding off;
                proxy_pass http://172.17.0.1:9001;
   }

      listen [::]:443 ssl ipv6only=on;
      listen 443 ssl;
      ssl_certificate /etc/ssl_certificates/wildap1.crt;
      ssl_certificate_key /etc/ssl_certificates/star1.key;
}

server {
    if ($host = vsk.ap.gov.in) {
        return 301 https://$host$request_uri;
    }


        listen       80;
        listen  [::]:80;        

        server_name vsk.ap.gov.in;
    return 404;


}

**Reply by PavitraIR** [reply]
@gprao08 Use this below ssl template and add the url and crt and key. save it and restart the container.

https://docs.google.com/document/d/1jLL1777LIAMfNu-B5kvZWgyi-nmAeivRjGPCsWQaDC4/edit

**Reply by gprao08** [reply]
@PavitraIR 
Plz provide the link access to gprao08@gmail.com

**Reply by gprao08** [reply]
@PavitraIR 
SSL template correct. But when I gave "**_sudo docker exec -it nginx_app bash_**"
It show the said docker stopped, I try to start the container it will not start. Screen shot attached for your reference.

![Uploading container.jpg…]()

**Comment by PavitraIR** [comment]
It's your server routing issue, please check with your devops team

**Reply by gprao08** [reply]
@PavitraIR @RNarmatha 
We have no routing issue. We open the ports. Even though we are unable to access the application. Could you plz support..
